### PR TITLE
add bazel v0.2.1 formula to support tensorflow syntaxnet

### PR DESCRIPTION
**Pull request by clarkbw**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/homebrew-core/blob/master/.github/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew-versions/pulls) for the same formula update/change?
- [x] Have you built your formula locally prior to submission with `brew install <formula>` (where `<formula>` is the name of the formula you're submitting)?
- [x] Does your submission pass `brew audit --strict --online <formula>` (after doing `brew install <formula>`)?
- [x] Does your submission adhere to Versions "Acceptable Formulae" [guidelines](https://github.com/Homebrew/homebrew-versions/blob/07b17dfb9f1/README.md#acceptable-formulae)?

-----

This version of bazel was inspired from a revision of brew/core bazel formula.
The tensorflow / syntaxnet project requires bazel versions 0.2.0 to 0.2.2b and not the latest 0.2.3 which is currently supported by homebrew. See https://github.com/tensorflow/models/tree/master/syntaxnet
Once syntaxnet updates their build system to the latest bazel this formula can and should be removed.

### CODE REVIEW

**Comment by ilovezfs** [comment]
@clarkbw 
```
==> brew audit homebrew/versions/bazel021 --strict --online
==> FAILED
Error: 2 problems in 1 formula
homebrew/versions/bazel021:
  * Stable: version 0.2.1 is redundant with version scanned from URL
  * The homepage is not reachable (curl exit code 22)
==> git checkout master -f
==> git reset --hard

```

**Comment by DomT4** [comment]
We're no longer generally accepting _point release_ formulae to Versions. Ideally, we only want Versions formulae that point to the major branch, so `node012 ` for example tracks the latest release in the `node012` release series. We wouldn't accept a `node01215` formula, etc.

We're aware Versions is currently a mess that makes this distinction confusing.

**Comment by clarkbw** [comment]
> accepting point release formulae to Versions.

Ok, I understand.  I can create a new repo for it and host it there.  Thanks!

**Comment by DomT4** [comment]
That's great. Thank you for understanding, appreciate that.

File: bazel021.rb [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  version "0.2.1"
+  sha256 "be964f98480ed258f249fc44c467a293e3a143aa750149b970da3d6719ba6ff1"
+  head "https://github.com/bazelbuild/bazel.git"
+
+  bottle :unneeded
+
+  depends_on :java => "1.8+"
+  depends_on :macos => :yosemite
+
+  conflicts_with "bazel", :because => "Differing version of same formula"
```
Comment (MikeMcQuaid) [inline bazel021.rb]: Move this above the `depends_on`: https://bot.brew.sh/job/Homebrew%20Versions%20Pull%20Requests/1561/version=yosemite/testReport/junit/brew-test-bot/yosemite/audit_homebrew_versions_bazel023___new_formula/

**Pull request closed without merging**